### PR TITLE
[expo-facebook] Add web support to Facebook

### PR DESCRIPTION
**Pull request by EvanBacon**
**HOLD** pending #7101

# How

- Add web support to `expo-facebook`.
- Add  method to log out of Facebook (iOS, Android, web) to test login on web.
- Add `getAccessTokenAsync` to tell if the app is authenticated or not.
- Change seconds to milliseconds to better accommodate JS rather than Objective-C.
- Change `initializeAsync` to accept an object instead. This is because AFAICT web doesn't have a notion of app names but does require the app version to be set. 
  - Added a deprecation method if the old signature was used `initializeAsync(appId, appName)` instead of `initializeAsync({ appId, appName })`.
- [x] Add guide for setting up web (this should be well tested).
- [x] Update docs to reflect new method signatures.
- [x] Update CHANGELOG to reflect breaking changes.

# Test Plan

- [docs preview](https://github.com/expo/expo/blob/@evanbacon/facebook/web-support-init/docs/pages/versions/unversioned/sdk/facebook.md)
- [x] Ensure everything works in NCL (manually)
- [x] Ensure everything works in Test Suite (manually)
  - [x] iOS 
  - [x] Android Emulator
  - [x] Android Device with Facebook installed and authenticated
  - [x] Web - log out seems to be [unreliable in JS FBSDK](https://stackoverflow.com/questions/22331373/facebook-js-api-cant-logout/25577384). Test Suite Facebook test reliably never resolves the second `logOutAsync()` call currently.

### CODE REVIEW

File: packages/expo-facebook/src/Facebook.ts [hunk 1]
```diff
@@ -113,3 +136,30 @@ export async function setAdvertiserIDCollectionEnabledAsync(enabled: boolean) {
   }
   return await ExponentFacebook.setAdvertiserIDCollectionEnabledAsync(enabled);
 }
+
+export async function getUserAsync(): Promise<any> {
```
Comment (EvanBacon) [inline packages/expo-facebook/src/Facebook.ts]: Should just demo this in the docs and not have it be part of the API

**Comment by minenkod** [comment]
Exciting! Awesome work!

**Comment by brentvatne** [comment]
@EvanBacon - should we close this or is it worth reviving it?

**Comment by hunght** [comment]
awesome work, could I know what time this will roll out?

**Comment by skymen** [comment]
When will this be ready you reckon?

**Comment by brentvatne** [comment]
@hunght @skymen - it's not being worked on actively, this was just a draft. you can use this to authenticate with fb on web: https://docs.expo.io/guides/authentication/#facebook

**Comment by haveamission** [comment]
@brentvatne - So to do that I'd probably want to use the Firebase Auth flow then?

**Pull request closed without merging**